add work for check

qwerq12311 opened the pull request.

qwerq12311 committed:
add allure report

pavel-ulan commented on pom.xml:
```diff
@@ -0,0 +1,124 @@
+            <groupId>org.openjfx</groupId>
+            <artifactId>javafx-fxml</artifactId>
+            <version>${javafx.version}</version>
+        </dependency>
```
**надо исправить**
серьезно, fx? решил запилить тесты в десктопном окошке?
я не знаю как так у тебя получилось, но лишних зависимостей в проекте быть не должно =)

pavel-ulan commented on src/main/java/CourierLoginEndpoint.java:
```diff
@@ -0,0 +1,22 @@
+
+public class CourierLoginEndpoint {
+
+    private static final String BASE_URL = "https://qa-scooter.praktikum-services.ru/";
```
**можно лучше**
смотри, у тебя BASE_URI повторяется везде, но при этом для всех классов он одинаковый. Как правило это говорит о том, что такого рода параметр можно вынести в конфигурацию всего проекта. И повторения кода не будет и исправлять если что надо будет только в одном месте

pavel-ulan replied:
Все еще актуально

pavel-ulan commented on src/main/java/CourierOrdersCountEndpoint.java:
```diff
@@ -0,0 +1,15 @@
+
+    public static Response getOrdersCountForCourier(int courierId) {
+        RestAssured.baseURI = BASE_URL;
+        String endpoint = ENDPOINT_COURIER_ORDERS_COUNT.replace(":id", String.valueOf(courierId));
```
есть такая штука, форматируемая строка называется =)
и велосипед изобретать не надо было бы

pavel-ulan commented on src/main/java/TestDataOrder.java:
```diff
@@ -0,0 +1,64 @@
+                "      \"color\": \"#D92B2C\"\n" +
+                "    }\n" +
+                "  ]\n" +
+                "}";
```
**надо исправить**
ух тыж ёлки-иголки 😮 
ну ты же используешь Gson... зачем вот эта вот простыня букафф?
Я догадываюсь, что у тебя заняло много времени составить эту строку, но это надо переделать.
Можешь объект смаршаллить в json, можешь map использовать, но ЭТО - нет

pavel-ulan commented on src/test/java/CourierTests.java:
```diff
@@ -0,0 +1,107 @@
+    @Test
+    @Description("Тест создания курьера с пустыми полями")
+    public void testCourierCreationWithEmptyFields() {
+        String emptyCourierJson = "{ \"login\": \"\", \"password\": \"\", \"firstName\": \"\" }";
```
вынеси их хотя бы в константы класса (а лучше сделай для них класс модель и создавай тестовые объекты)

pavel-ulan commented on src/test/java/OrderAcceptanceEndpointTest.java:
**надо исправить**
а это что за артефакт? (не оставляй закомменченные куски кода в работе)

pavel-ulan commented on src/test/java/OrderCreationEndpointTest.java:
```diff
@@ -0,0 +1,103 @@
+
+                // Обязательные поля заполнены, указаны оба цвета: BLACK и GREY
+                {new OrderDetails("Ibragim", "Kandibober", "123 Afganistan", "Nearest Station", "1111-22", 4,
+                        "2023-07-30T21:00:00.000Z", "FASTEEEEER", Arrays.asList("BLACK", "GREY"))}
```
**хорошо**
ну вот хотя бы вот так. есть объект, структура, данные в этом объекте. вот так - хорошо

pavel-ulan commented on src/test/java/OrderCreationEndpointTest.java:
```diff
@@ -0,0 +1,103 @@
+        System.out.println("Заказ создан. Номер отслеживания (track): " + track);
+
+        // Allure: добавление шага в отчет
+        Allure.step("Заказ с номером отслеживания (track): " + track + " успешно создан");
```
**можно лучше**
вообще это работает не так. Молодец, конечно, что нашел обходной путь, но вообще к rest-assured просто подключается allure-rest-assured плагин и он отлично транслирует все действия в виде шагов в отчете (всегда изучай внимательно документацию к используемым библиотекам и фреймворкам).

pavel-ulan reviewed (changes requested):
Ну, явно есть над чем поработать. Хотя в целом ничего непоправимого нет, так что думаю больших проблем для тебя это не будет представлять

qwerq12311 committed:
1. удалил ненужные зависимости
2. вынес base_url в отдельный класс
3. использовал форматируемую строку и обновил код
4. изменил полностью конструктор заказа
5. создал еще один метод для проверки кейса с пустым логином и паролем. обновил тесты на этом основании
6. удалил ненужный тест
7. подключил allure-rest-assured. обновил pom. переписал OrderCreationEndpointTest

summary : все замечания учтены и исправлены

pavel-ulan commented on src/test/java/CourierTests.java:
```diff
@@ -0,0 +1,126 @@
+import static org.junit.Assert.*;
+
+@Epic("Тестирование API")
+@Feature("Курьеры: Создание и авторизация")
```
во, красивенько

pavel-ulan commented on src/test/java/CourierTests.java:
```diff
@@ -0,0 +1,126 @@
+
+        // Дополнение для проверки успешного завершения теста
+        assertTrue("Тест успешной авторизации курьера завершен успешно", true);
+        System.out.println("Тест успешной авторизации курьера завершен успешно");
```
**можно лучше**
не используй SOUT в финальной версии кода. SOUT это вывод в консоль, там где ты будешь гонять тесты консоли нет. Там есть ЛОГ. Если хочешь что-то вывести, используй логгирование. SOUT допустим только во время отладки

pavel-ulan commented on src/test/java/TestGetOrders.java:
```diff
@@ -0,0 +1,50 @@
+        return testDataOrder.getOrdersData();
+    }
+
+    @Step("тест-тест-тест")
```
😆

pavel-ulan commented on src/main/java/OrderAcceptanceEndpoint.java:
```diff
@@ -0,0 +1,21 @@
+
+public class OrderAcceptanceEndpoint {
+
+    private static final String BASE_URL = "https://qa-scooter.praktikum-services.ru/";
```
**можно лучше**
непоняяятно... ты вынес URI в константу конфиг класса. Зачем дублировать ее тут? Обращайся напрямую к той константе.

pavel-ulan reviewed (approved):
Да, стало получше. Все еще надо по мелочам допилить (заменить системные выводы, константы поправить, все еще можно уменьшить влияние тестовых данных (встречается "легкий хардкод")). Но в целом - теперь это похоже на рабочий проект, молодец =)
